src/parser/queries/QueryBuilder.ts

```diff
@@ -26,50 +34,114 @@ export class QueryBuilder {
+     * @return The query, or undefined if the file extension / language of the source file is not supported.
+     */
+    build(): Query | undefined {
         let queryString = "";
+        if (this.language !== Languages.Unsupported) {
+            if (this.statements.length > 0) {
+                queryString = this.getStatementsQuery();
+            }
+
+            dlog("------------- Start Query: --------------");
+            dlog(queryString);
+            dlog("-----------------------------------------");
 
-        if (this.statements.length > 0) {
-            queryString = this.getBruteForcedStatementsQuery();
+            return new Query(this.treeSitterLanguage, queryString);
         }
+    }
 
-        dlog("------------- Start Query: --------------");
-        dlog(queryString);
-        dlog("-----------------------------------------");
+    /**
+     * Generates a query string for finding the currently set statements in code of the currently set language.
+     * Returns an empty query if the language is not supported.
+     * @return A query string.
+     * @private
+     */
+    private getStatementsQuery() {
+        const statementQueries: string[] = [];
 
-        return new Query(this.treeSitterLanguage, queryString);
+        for (const statement of this.statements) {
+            // Is "activatedFor" still necessary? I have only found one node with the activated_for_languages filed.
+            // However, it seems helpful for some (e.g. debugging) purposes,
+            // so I kept it and introduced "applicableFor" along it.
+            if (statement.applicableFor(this.language) && statement.activatedFor(this.language)) {
+                statementQueries.push(statement.toQuery());
+            }
+        }
+        return statementQueries.join("\n");
     }
 
-    // This is not necessary anymore
-    // due to the mapped node types (/resources/node-types-mapped.config)
-    // and the information per expressions
-    // in which language it is included
-    // Thus, we do not have to figure out
-    // if the expression is included in a language by query the expression
-    // and catch an exception if it is not included.
-    private getBruteForcedStatementsQuery() {
-        const availableStatements: string[] = [];
-
-        for (const statementCandidate of this.statements) {
-            if (!statementCandidate.activatedFor(this.fileExtension)) {
-                continue;
+    /**
+     * ONLY FOR DEBUGGING PURPOSES!
+     * This lets you compare the result of the new approach for getStatementsQuery which relies on the "language"
+     * field of the expression mappings to the old, very slow bruteforce approach.
+     * @param returnBruteforceResult Whether to return the result of the bruteforce approach or the new approach.
+     * @return A query string.
+     * @private
+     */
+    private debugCompareBruteforceToMapped(returnBruteforceResult: boolean) {
```

Comment:
I know it is still a draft one, but don't forget to resolve this.

Reply:
Yes, we might remove this once the issue with the node mappings is resolved. Until then, it is quite helpful to compare the results of the old and the new approach if necessary.

I don't remember if we discussed how to proceed on this issue. Should we first implement a new procedure for updating the grammars and generate a new node mappings file which fixes the issues that occured here? Or should I try to fix the existing file until the output from the dependency analysis equals that of the test case (minus the thing mentioned in #26) ?

Reply:
Debug bruteforce function removed